common.go

```diff
@@ -160,35 +160,34 @@ const (
 	// Rest of these are reserved by the TLS spec
 )
 
-// Hash functions for TLS 1.2 (See RFC 5246, section A.4.1)
+// Public key algorithm type.
+type pubkeyType uint8
+
 const (
-	hashSHA1   uint8 = 2
-	hashSHA256 uint8 = 4
-	hashSHA384 uint8 = 5
+	keyRSA pubkeyType = iota
+	keyECDSA
 )
 
-// Signature algorithms for TLS 1.2 (See RFC 5246, section A.4.1)
+// A generic signature algorithm, not including any hash function.
+type signatureType uint8
+
 const (
-	signatureRSA   uint8 = 1
```

Comment:
I think I missed something. Now signatureRSAPKCS1=0, RSAPSS=1, and signatureECDSA=2?

Reply:
Correct, the exact values do not exactly matter since the `signatureType` is not sent over the network, it is used only internally to track what signature verification method needs to be called.

In TLS 1.2 the SignatureAndHash (SAH) type consisted of two fields, these were merged in TLS 1.3 into a single 16-bit number. The hash type and the SAH types have been removed in the previous patch, this one removed the signature type from SAH as well.

Reply:
Make sense. I only briefly touched those code before, I didn't know the change in progress.